crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -1431,31 +1441,37 @@ impl<'src> Parser<'src> {
                 }
             }
             _ => {
+                // grammar: `group`
+                if parsed_expr.expr.is_starred_expr() {
+                    self.add_error(
+                        ParseErrorType::OtherError(
+                            "starred expression cannot be used here".to_string(),
+                        ),
+                        &parsed_expr,
+                    );
+                }
+
```

Comment:
This is for `(*expr)` which isn't allowed anywhere. This is the reason we check for whether the starred expression is parenthesized or not in the previous diff chunk where we raise `IterableUnpackingInComprehension`.